docs/sdg/sdg-refactor.md

```diff
@@ -0,0 +1,136 @@
+2. User installs the SDG library only.  They want to invoke any of the following as a library call from code they write:
+    - 2.1. They have a taxonomy with some new seed data they added to it.  They want to run SDG only without any postprocessing.
+    - 2.2. They have a set of seed context/question/answer tuples and optionallly a set of document chunks.  They want to run SDG only without any postprocessing.
+
+If I understand the guidance from our PM (William Caban), the flows we are being asked to support here are 1.1., 1.2., 1.3, 1.3.1, and 2.2.  However, I am not sure that I understand the guidance.  Are we confident that we _do_ want to support those flows and do not want to support any of the others?  More specifically, do we think the users who want 2.2 might be satisfied with 1.6 instead?  Both of those are SDG-only flows, but the latter is more developer focused and the former is more business-user focused.  Also, are we really confident that all of the "SDG only" customers really want to do their own document chunking?  Might there be some customers that have seed context/question/answer tuples and _documents_ and want us to chunk the documents for them?
+
+Also note, that for all of the flows above that start with "they have a taxonomy", there are open questions around the complexity of the taxonomy format.  Some users want to be able to provide seed data and documents without explicitly or implicitly extending some sort of base taxonomy.  If we had something simpler than the existing taxonomy format that still included some way to specify context/question/answer tuples and references to full documents (i.e., not _chunks_), would we even call that a taxonomy or would it be something else?  If we call it something else, then some or all of the flows that take in a taxonomy might also be applicable for that something else.  For the remainder of this document, however, we will assume that any sort of simplified/easier-to-use variant of a taxonomy will also be called a "taxonomy".
+
+## Question 2: What should the commands be in the CLI?
+
+One way to support both 1.3.1 and 1.3.2 would be to have separate CLI commands for the preprocessing, core SDG, and postprocessing step .  Alternatively, a single CLI command that does all of these and also saves the outputs of preprocessing to disk would support 1.3.1 but _not_ 1.3.2.  Even if we only want to support 1.3.1, having separate CLI commands for each step might be desirable because it is just more intuitive that if a user wants to save the outputs of preprocessing to disk to have a command to do that instead of having it be a "side effect" of an omnibus SDG command.  Here is a rough outline of what the separate commands would be:
+
+- `ilab data prep` would handle all the preprocessing (the first three bullets in the Context section above, plus any additional preprocessing we add in the future).
```

Comment:
As we discussed, I like the split of these commands, the modularity makes sense to me!

Reply:
I too like splitting this out, although I'm not entirely sure if this is the right dimension or if conversion of documents with Docling should be entirely separate from traversing a taxonomy, extracting data from qna.yamls, and fetching the referenced knowledge. To put it more specifically, we may want to end up with something like `ilab document convert` that converts a set of input documents to a set of docling jsons or markdown that may be subsequently hand-edited, committed to a git repository, or something else. In the spirit of this document, we probably don't have to decide this here and now. But, document conversion may have a separate iteration loop from a user's point of view, and not just be something that happens automatically during SDG.